src/utilities.lisp

```diff
@@ -156,6 +157,18 @@
          (nreverse out)
          result)))))
 
+(defun take (n list)
```

Comment:
these functions work for any sequence

Reply:
Restricting their type with a declaration in a local function allows better type inference and checking, plus potentially compiling away some dispatch.

Reply:
what you say is true yet i don't see many calls to LIST-LENGTH in our code

Reply:
😬

Beware, it returns NIL on circular lists.